lib/minitest/assertions.rb

```diff
@@ -50,27 +50,32 @@ def self.diff= o
       @diff = o
     end
 
+    ##
+    # Returns things to diff [expect, butwas], or [nil, nil] if nothing to diff.
+    def things_to_diff(exp, act)
+      expect = mu_pp_for_diff exp
+      butwas = mu_pp_for_diff act
+      need_to_diff =
+          (expect.include?("\n")    ||
+              butwas.include?("\n")    ||
+              expect.size > 30         ||
+              butwas.size > 30         ||
+              expect == butwas)        &&
+              Minitest::Assertions.diff
+      need_to_diff ? [expect, butwas] : [nil, nil]
```

Comment:
this might just be me... but I would do `need_to_diff && [expect, butwas]` and let ruby do the rest